src/pages/track/TrackPage.tsx

```diff
@@ -35,13 +39,16 @@ export const TrackPage = ({team}: TrackProps) => {
     if (loading) return (<Loading text={"Cargando..."}/>)
 
     return (
-        <div>
-            <form onSubmit={handleSubmit}>
-                <p>{checkpoint.question}</p>
-                <TextInput id="answer" placeholder="----" onChange={setAnswer}/>
-                <br/>
-                <button type="submit">Enviar</button>
-            </form>
-        </div>
+        <>
+            <LogOut text={"pa tu casa"} logout={logout({removeTeam : removeTeam, setTeam : setTeam})}/>
```

Comment:
I think that here we want to call a _use case_ for calling the `logout`. You are using your `setTeam` as callback. However can be simple if you split the _chain of responsibility_. Your component will handle itself cleaning the team, and the `logout` _use case_ will be the one that removes the team at _storage system_.

Reply:
I do not understand what are you talking about. I want to inject the dependencies of the storage because it can change in the future but also, I need to inject the `setTeam` because the team is created at `App.tsx`.  So, I send all the deps from app > tackPage > logout.